hanging_threads.py

```diff
@@ -75,62 +147,28 @@ def frame2string(frame):
 
 
 def thread2list(frame):
-    """Return list with string frame representation of each fame of thread"""
+    """Return list with string frame representation of each frame of thread"""
     l = []
     while frame:
         l.insert(0, frame2string(frame))
         frame = frame.f_back
     return l
 
 
-def monitor(seconds_frozen, tests_per_second):
-    """Monitoring thread function
+def log_hanged_thread(thread_id, frame):
+    """Print the stack trace of the deadlock after hanging `seconds_frozen`"""
+    write_log('Thread {0} hangs '.format(thread_id), ''.join(frame))
 
-    Checks if thread is hanging for time defined by
-    seconds_frozen parameter in frequency
-    of test_per_second parameter
-    """
-    current_thread = threading.current_thread()
-    self = get_ident()
-    old_threads = {}
-    while not current_thread.is_stopped():
-        time.sleep(1. / tests_per_second)
-        now = time.time()
-        then = now - seconds_frozen
-        frames = sys._current_frames()
-        new_threads = {}
-        for frame_id, frame in frames.items():
-            new_threads[frame_id] = thread2list(frame)
-        for thread_id, frame_list in new_threads.items():
-            if thread_id == self: continue
-            if thread_id not in old_threads or \
-               frame_list != old_threads[thread_id][0]:
-                new_threads[thread_id] = (frame_list, now)
-            elif old_threads[thread_id][1] < then:
-                print_frame_list(frame_list, thread_id)
-            else:
-                new_threads[thread_id] = old_threads[thread_id]
-        old_threads = new_threads
 
+def log_awaked_thread(thread_id):
+    write_log('Thread {0} awaked'.format(thread_id))
 
-def print_frame_list(frame_list, frame_id):
-    """Print the stack trace of the deadlock after hanging `seconds_frozen`"""
-    sys.stderr.write('-' * 20 + 
-                     'Thread {0}'.format(frame_id).center(20) +
-                     '-' * 20 +
-                     '\n' + 
-                     ''.join(frame_list))
 
+def log_died_thread(thread_id):
+    write_log('Thread {0} died  '.format(thread_id))
 
-def start_monitoring(seconds_frozen=SECONDS_FROZEN,
-                     tests_per_second=TESTS_PER_SECOND):
-    """Start monitoring threads
 
-    seconds_frozen - How much time should thread hang to activate printing stack trace - default(10)
-    tests_per_second - How much tests per second should be done for hanging threads - default(10)
-    """
-    thread = StoppableThread(target=monitor, args=(seconds_frozen,
-                                                   tests_per_second))
-    thread.daemon = True
-    thread.start()
-    return thread
+def write_log(title, message=''):
```

Comment:
I thought somebody may want to replace this function with own implementation to write messages to another destination in another format.